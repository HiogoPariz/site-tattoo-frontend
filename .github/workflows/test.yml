name: Test workflow

on:
  pull_request:
    branches: [staging]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Check out
        uses: actions/checkout@v2

      - name: Set up Node 14
        uses: actions/setup-node@v3
        with:
          node-version: '14'
          cache: 'yarn'
          cache-dependency-path: yarn.lock

      - name: Install dependencies
        run: yarn install --frozen-lockfile

      - name: Test with jest
        run: yarn test --ci --maxWorkers=100%
