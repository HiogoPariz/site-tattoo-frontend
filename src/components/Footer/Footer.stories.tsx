import React from 'react'
import { Meta, Story } from '@storybook/react'

import { Footer } from './Footer'

const meta: Meta = {
  title: 'Components/Footer',
  component: Footer
}

export default meta
const Template: Story = args => <Footer {...args} />

export const Default = Template.bind({})
