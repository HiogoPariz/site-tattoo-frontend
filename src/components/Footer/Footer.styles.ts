import styled from 'styled-components'

export const FooterContainer = styled.div`
  display: grid;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.black};
  grid-area: footer;
  padding: ${({ theme }) => theme.spacings.normal};
`

export const GroupContainer = styled.ul`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.spacings.normal};
`
export const GroupTitle = styled
export const GroupLink = styled.li``
