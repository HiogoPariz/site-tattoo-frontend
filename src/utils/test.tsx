/* eslint-disable import/export */
import React from 'react'
import { render } from '@testing-library/react'
import theme from '@/styles/theme'
import { ThemeProvider } from 'styled-components'

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
