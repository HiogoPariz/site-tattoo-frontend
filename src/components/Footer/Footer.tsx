import { NavLink } from '@/types'
import React from 'react'
import { FooterContainer, GroupContainer, GroupLink } from './Footer.styles'
import Link from 'next/link'

export type FooterGroup = {
  name: string
  links: NavLink[]
}

export type FooterProps = {
  groups: FooterGroup[]
}
export const Footer: React.FC<FooterProps> = ({ groups }) => {
  return (
    <FooterContainer>
      {groups.map(group => (
        <GroupContainer key={`footer-group-${group.name}`}>
          <h2>{group.name}</h2>
          {group.links.map(link => (
            <GroupLink
              key={`footer-link-${group.name}-${link.label}-${link.to}`}
            >
              <Link href={link.to}>{link.label}</Link>
            </GroupLink>
          ))}
        </GroupContainer>
      ))}
    </FooterContainer>
  )
}
