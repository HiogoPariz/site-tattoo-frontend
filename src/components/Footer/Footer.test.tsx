import React from 'react'
import { render } from '@/utils/test'
import { Footer } from './Footer'

describe('<Footer />', () => {
  it('should render', () => {
    const { container } = render(<Footer />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toBeInTheDocument()
  })
})
