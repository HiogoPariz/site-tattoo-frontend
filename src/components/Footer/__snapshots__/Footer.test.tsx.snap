// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c0 {
  display: grid;
  width: 100%;
  background-color: #000000;
  grid-area: footer;
  padding: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <h2>
      rotas
    </h2>
    <li
      class="Footerstyles__GroupLink-sc-z3hj90-2"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="Footerstyles__GroupLink-sc-z3hj90-2"
    >
      <a
        href="/teste"
      >
        Teste
      </a>
    </li>
  </ul>
</div>
`;
